File: sheet 4 , M. Subsequence String/main.py
def find_hello_subsequence(s):
    target = "hello"
    i = 0  
    for char in s: #ahhellllloou
        if char==target[i]:
            i+=1
            if i==len(target):
                return "YES"

    return "NO"

S = input().strip()

result = find_hello_subsequence("ahhellllloou")
print(result)

File: sheet 4 , M. Subsequence String/main.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function findHelloSubsequence(s) {
    const target = "hello";
    let i = 0;  // Index for the target string "hello"

    for (const char of s) {
        if (char === target[i]) {
            i++;
            if (i === target.length) {
                return "YES";
            }
        }
    }

    return "NO";
}

rl.question("", (S) => {
    const result = findHelloSubsequence(S.trim());
    console.log(result);
    rl.close();
});
